pkg/kubelet/secret/secret_manager.go

```diff
@@ -209,8 +209,12 @@ type cachingSecretManager struct {
 }
 
 func NewCachingSecretManager(kubeClient clientset.Interface) (Manager, error) {
+	// TODO: Currently apiserver completely cannot keep up with 1 minute ttl in
+	// 5000-node clusters. So before we optimize apiserver and/or etcd, increase
+	// the TTL to 2 minutes (which is a lot, but...)
+	ttl := 2 * time.Minute
```

Comment:
Long term this should be watch based (watch all secrets I'm allowed to watch). But supporting that in apiserver is a large project which won't happen in 1.6. So we need some short term solution.
I agree that this suggestion is not ideal, but I also don't think it's that bad, because:
- this is only "read" capability
- it's about the resource that it already can read (node) [though potentially not for all objects]
- it will be super "local" change (so removing this in the future will be easy)
- it is supposed to be temporary solution until apiserver will be able to do what it should already be able to do
 So maybe that's not that bad?

 [Though if you have better suggestion, I'm happy to hear it. I just a bit sceptical about the flag, as  then configuration became painful, not even saying about changes in cluster size...]